src/components/authentication/SigninXrayWrapper.vue

```diff
@@ -0,0 +1,264 @@
+        <v-btn
+          color="primary"
+          text
+          v-bind="attrs"
+          @click="snackBar.enabled = false"
+        >
+          Close
+        </v-btn>
+      </template>
+    </v-snackbar>
+  </v-container>
+</template>
+<script>
+import axios from "axios";
+import dayjs from "dayjs";
+import xrayIntegrationHelper from "../../integrations/XrayIntegrationHelpers";
+export default {
+  name: "SigninXrayWrapper",
+  props: {
+    configItem: {
+      type: Object,
+      default: () => {},
+    },
+    credentialItems: {
+      type: Object,
+      default: () => {},
+    },
+    prevRoute: {
+      type: Object,
+      default: () => {},
+    },
+  },
+  watch: {
+    configItem: function (newValue) {
+      this.config = newValue;
+    },
+    credentialItems: function (newValue) {
+      this.credentials = newValue;
+    },
+    prevRoute: function (newValue) {
+      this.previousRoute = newValue;
+    },
+  },
+  data() {
+    return {
+      config: this.configItem,
+      credentials: this.credentialItems,
+      previousRoute: this.prevRoute,
+      client_id: "",
+      showClientId: false,
+      client_secret: "",
+      showClientSecret: false,
+      loading: false,
+      valid: true,
+      rules: {
+        client_id: [
+          (v) =>
+            !!v ||
+            this.$i18n.t("message.client_id") +
+              this.$i18n.t("message.is_required"),
+        ],
+        client_secret: [
+          (v) =>
+            !!v ||
+            this.$i18n.t("message.client_secret") +
+              this.$i18n.t("message.is_required"),
+        ],
+      },
+      snackBar: {
+        enabled: false,
+        message: "",
+      },
+    };
+  },
+  computed: {
+    currentTheme() {
+      if (this.$vuetify.theme.dark) {
+        return this.$vuetify.theme.themes.dark;
+      } else {
+        return this.$vuetify.theme.themes.light;
+      }
+    },
+  },
+  methods: {
+    back() {
+      if (this.previousRoute.path.includes("setting")) {
+        this.$router.push({ path: this.previousRoute.path });
+      } else {
+        this.$router.back();
+      }
+    },
+    async signIn() {
+      const isValid = this.$refs.form.validate();
+      if (isValid) {
+        this.postLogin({
+          xray: {
+            client_id: this.client_id,
+            client_secret: this.client_secret,
+          },
+        });
+      }
+    },
+    async postLogin(data) {
+      console.log(data); //TODO - Abstract this to helper.
+
+      let header = {
+        headers: {
+          "Content-Type": "application/json",
+        },
+      };
+
+      await axios
+        .post(
+          `https://xray.cloud.getxray.app/api/v2/authenticate`,
+          data.xray,
+          header
+        )
+        .then((user) => {
+          const date = dayjs().format("YYYY-MM-DD HH:mm:ss");
+          const xrayData = {
+            ...data.xray,
+            loggedInAt: date,
+            auth_token: user.data,
+          };
```

Comment:
@Zubs yes I see the problem here.  I guess the best we could do is try to use their JIRA user if they are also logged into JIRA.  This will also be a problem with other JIRA-based tools that we integrate with so that may be our best option.